evaluator/builtin_string.go

```diff
@@ -684,11 +681,7 @@ func builtinChar(args []types.Datum, ctx context.Context) (d types.Datum, err er
 		}
 	}
 
-	resultBytes, err = codec.ConvertInt64ToBytes(intSlice, binary.BigEndian, 4)
-	if err != nil {
-		return d, errors.Trace(err)
-	}
-	resultStr = hack.String(resultBytes)
+	resultStr = hack.String(convertInt64ToBytes(intSlice))
```

Comment:
IMHO, I think stringHeader.Data holds the reference of data array, so the real data won't be recycled by GC. I tested with the following code, the result is that the value of s still exists. Is this correct?
```
	s := f()
	for i := 0; i < 100; i++ {
		runtime.GC()
	}
	fmt.Println(s)
```